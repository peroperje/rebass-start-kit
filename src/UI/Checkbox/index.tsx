import * as forms from "@rebass/forms";

export type CheckboxProps = forms.CheckboxProps;

const Checkbox = forms.Checkbox;

export default Checkbox;
