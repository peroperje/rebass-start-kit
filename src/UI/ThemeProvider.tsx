import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
