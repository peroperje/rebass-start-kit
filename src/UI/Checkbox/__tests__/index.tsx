import React from "react";
import render from "react-test-renderer";

import Checkbox from "../index";
import ThemeProvider from "../../ThemeProvider";

describe("Checkbox", () => {
  it("Default snapshot", () => {
    const tree = render
      .create(
        <ThemeProvider>
          <Checkbox>Test</Checkbox>
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
