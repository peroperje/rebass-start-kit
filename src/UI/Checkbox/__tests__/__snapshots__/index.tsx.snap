// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Default snapshot 1`] = `
.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  width: 24px;
  height: 24px;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 4px;
  color: #dddddf;
}

input:checked ~ .emotion-1 {
  display: block;
}

input:checked ~ .emotion-1 {
  color: #07c;
}

input:focus ~ .emotion-1 {
  color: #07c;
  background-color: hsla(205,100%,40%,0.125);
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 24px;
  height: 24px;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 4px;
  color: #dddddf;
}

input:checked ~ .emotion-2 {
  display: none;
}

input:checked ~ .emotion-2 {
  color: #07c;
}

input:focus ~ .emotion-2 {
  color: #07c;
  background-color: hsla(205,100%,40%,0.125);
}

<div
  className="emotion-3"
>
  <input
    className="emotion-0"
    type="checkbox"
  >
    Test
  </input>
  <svg
    aria-hidden="true"
    className="emotion-1"
    fill="currentcolor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="emotion-2"
    fill="currentcolor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
    />
  </svg>
</div>
`;
